Added the vonage status panel plugin

TomerAmirV opened the pull request:
I created a new panel plugin that is similar to the single stat panel, but it allows to gather data from multiple queries, and decide the status of the measured target according to thresholds of every query.

There are screenshots attached, and also a description in the readme file.

Thanks!

daniellee commented:
Thanks for submitting this! We had a busy week last week with GrafanaCon but now I'm back in action. I'll get back to you soon with feedback.

TomerAmirV commented:
Fixed merge conflicts.

daniellee commented:
Hi!

Sorry about the delay, still catching up on my backlog after GrafanaCon! 

There are two parts to my feedback, the metadata for publishing the plugin and then the plugin code.

### Metadata

The metadata is mostly fine except for two small things. 

1. In the plugin json, could you remove the logos section rather than having empty values in the plugin.json files in [src](https://github.com/Vonage/Grafana_Status_panel/blob/master/src/plugin.json#L13-L16) and [dist](https://github.com/Vonage/Grafana_Status_panel/blob/master/dist/plugin.json#L13-L16).

2. We use Readme.md on the Grafana.net site and we don't allow html in Markdown right now. Can you find another way to link in the following images (might have to resize the images):

```
<img src="https://github.com/Vonage/Grafana_Status_panel/blob/develop/src/img/ok.png?raw=true" height="100px" />
<img src="https://github.com/Vonage/Grafana_Status_panel/blob/develop/src/img/warning.png?raw=true" height="100px" />
<img src="https://github.com/Vonage/Grafana_Status_panel/blob/develop/src/img/error.png?raw=true" height="100px" />
```

### Feedback on the plugin

1. I noticed that it works with InfluxDB but not Graphite (there is an alias function in Graphite but not an alias field on target). Can you write in the readme which databases the plugin supports.
2. What does the remove prefix field do? (The code looks like it has been commented out: https://github.com/daniellee/Grafana_Status_panel/blob/master/src/status_ctrl.js#L181)
3. Could you document how having multiple values and multiple thresholds work. Just a sentence or two explaining that they are combined (e.g. all values must be under the threshold for it to be green, if one is over then it is red)
4. It took me a while to figure out that `Name` had nothing to with the aliases and was just a title. Maybe something like Panel Title would be a better name for Name.
5. Really nice that it just picks up the multiple queries automatically and shows them on the options panel.
6. Also nice that when you add a new query, that it takes the default threshold values from the previous query.
7. Get a JavaScript error when I create a new panel and it tries to [set the display name](https://github.com/Vonage/Grafana_Status_panel/blob/master/src/status_ctrl.js#L59-L60). An if statement guarding the render function if no values have been set yet would stop that.
8. I also got an JavaScript error when testing against Graphite Data as there was no alias field on target [here](https://github.com/Vonage/Grafana_Status_panel/blob/master/src/status_ctrl.js#L31)

Great work on the plugin! Thanks again for submitting it, I can see how it could be very useful for server stats. 

For the documentation, you can either have it in the readme or embedded in the plugin with a popover:
![image](https://cloud.githubusercontent.com/assets/434655/21140904/ad91d4b2-c13a-11e6-92ee-a630468559f0.png)



or maybe a toggle:

![image](https://cloud.githubusercontent.com/assets/434655/21140793/2f715abc-c13a-11e6-9482-74faab7e712d.png)

TomerAmirV commented:
Thanks for the comments!
I will work on them and update you here as soon as I'm done.

TomerAmirV commented:
@daniellee Done with the issues you reported.
About Graphite, currently we do not support it because we don't use it with this panel.
So I added to the documentation a section that says we only support InfluxDB currently, and I created an issue in our repository, and the project is open source, so if I see demand for this feature I might do it, or if someone wants it, they could contribute 😄 

Hope there is nothing else.

daniellee commented:
That was fast! Two things to fix:
1. Update the plugin json as mentioned above:

> In the plugin json, could you remove the logos section rather than having empty values in the plugin.json files in src and dist.

2. The path for the screenshot image is wrong in the plugin.json (src and dist). It should be relative to the where the plugin.json file is.

    `{"name": "Environment Snapshot", "path": "src/img/environment_snapshot.png"}`

    should be:

    `{"name": "Environment Snapshot", "path": "img/environment_snapshot.png"}`

TomerAmirV commented:
@daniellee Sorry for that. Made the fixes in the `dist` folder, and got reverted one I compiled the code.

I think it's done now.

daniellee commented:
No worries, I have to build a linting tool for this soon so that you can get direkt feedback on metadata errors. It's all a bit too manual at the moment.

Merged and published: https://grafana.net/plugins/vonage-status-panel

Thanks for submitting this! Do you need help with an icon? I could ask our designer to fix one for you.

TomerAmirV commented:
Hmmm... I asked our designer to make one, but she is very busy. If we won't have an icon ready in the next few days, I might take you on your offer 😄

TomerAmirV commented:
@daniellee 
I spoke with our designer, and it seems like she won't have time to make an icon for us. Could your designer make something? Thanks!

daniellee commented:
I'll ask him. Not sure if he'll have time the week before Christmas though!

daniellee commented:
@TomerAmirV here's an icon made by @bulletfactory (Thanks Trent!)

![icon_status_panel](https://cloud.githubusercontent.com/assets/434655/21355861/eb804174-c6cf-11e6-9006-1a7f387e8a18.png)

A zip with a png and svg versions:

[icon_status_panel.zip](https://github.com/grafana/grafana-plugin-repository/files/664043/icon_status_panel.zip)

If you can add it to your github repo, link it in the plugin.json and send me the commit hash, then I'll get it published straight away.

TomerAmirV commented:
@daniellee @bulletfactory 
Awesome Icon! I added it to our repository, and updated the plugin json.
The commit hash is: d87978944e19edc0ec82b3a24696cba74c944a42

Thanks a lot!

TomerAmirV commented:
@daniellee Sorry!

fixed, new hash: b1573ac0c238ddbb170a0f55af76dc62fb90d550

daniellee commented:
Great! Updated and published.

TomerAmirV commented:
Awesome!
Thanks!

TomerAmirV commented:
@daniellee 
We found a small bug. The link to our license file if broken in your site since it's relative.
I changed it to an absolute link, and we would be happy if you could update your site as well.

The hash is: 4754eb15e8fc6938afa195be03cd0843e2834071

Thanks!

daniellee commented:
@TomerAmirV you forgot to update the link in the README.md in the dist directory. That's the one Grafana uses.

TomerAmirV commented:
@daniellee 
Thanks! I did not realize that. here is the fixed hash: 78c1d8b667c299d60fd65842d0604a06da74333c

daniellee commented:
Thanks! Published.
